src/Engines/thinEngine.ts

```diff
@@ -2886,6 +2886,10 @@ export class ThinEngine {
             url = this._transformTextureUrl(url);
         }
 
+        if (originalUrl !== url) {

```

Comment:
Just a thought on this, but if there were a second variable like `transformedUrl`, then subsequent code could just look for `transformedUrl || originalUrl` which might make it easier to reason about this part of the code.

Reply:
sorry I merged before seeing your comment :-( @Popov72  do not hesitate to make another PR if needed... Sorry again

Reply:
Not sure there's a big difference between having `originalUrl / url` and `url / transformedUrl` (if that's what your comment is about)?

Reply:
My thinking is that `url` is being used as a source of truth, but it is being overwritten by the transformed URL, so I was thinking it's better to have unique variables with descriptive names so that there are no side effects, which is what caused this bug to slip by.